src/us/kbase/common/service/JsonTokenStream.java

```diff
@@ -276,6 +276,8 @@ public JsonTokenStream setRoot(List<String> root) throws JsonParseException, IOE
 			lastToken = nextToken();
 		List<String> ret = new ArrayList<String>();
 		int size = path.size() - 1;
+		// this is a bug. If the path is empty and this method is called 
+		// you'll get an index exception
```

Comment:
Didn't feel like fixing it?

Reply:
Not in this PR. Maybe later. The actual bug doesn't seem to occur in the wild.

Reply:
And if it does occur it'll be immediately obvious and the json processing will stop as it'll throw a RuntimeException.